### PR TITLE
magic-wormhole-rs: 0.3.0 -> 0.5.0

### PR DESCRIPTION
**Pull request by dywedir**
###### Description of changes

https://github.com/magic-wormhole/magic-wormhole.rs/releases/tag/0.4.0
https://github.com/magic-wormhole/magic-wormhole.rs/releases/tag/0.5.0

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by piegamesde** [comment]
Please also add the alias like in #174628. I think that the full name will making it easier to find the package via the search (especially since we don't really have a tagging system), but people that know the application are likely to just try `nix-shell -p wormhole-rs` given that this is the binary name.

Also, I suggest backporting this change to the coming release, because the previous versions have a compatibility bug with the other client implementations.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -27937,7 +27937,9 @@ with pkgs;
 
   magic-wormhole = with python3Packages; toPythonApplication magic-wormhole;
 
-  magic-wormhole-rs = callPackage ../tools/networking/magic-wormhole-rs/default.nix { };
+  magic-wormhole-rs = callPackage ../tools/networking/magic-wormhole-rs/default.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:27940]:
```suggestion
  magic-wormhole-rs = callPackage ../tools/networking/magic-wormhole-rs {
```

**Comment by dywedir** [comment]
> Please also add the alias like in #174628. I think that the full name will making it easier to find the package via the search (especially since we don't really have a tagging system), but people that know the application are likely to just try `nix-shell -p wormhole-rs` given that this is the binary name.

I think we should be looking at the name of the package in other repositories, but not at the name of the binary itself.
For this package, we don't have enough statistics yet (https://repology.org/project/magic-wormhole-rs/versions), so I would suggest postponing this proposal, maybe?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (dywedir) [reply pkgs/top-level/all-packages.nix:27940]: Done, thanks!

**Comment by piegamesde** [comment]
Both package names are equally fine. The details of the decision depend on distribution specific details, but in general it's a tradeoff between a shorter, more convenient name and one that will yield better search results. Nixpkgs is in the unique situation of allowing package for this exact purpose and I'd like to make use of them here.

**Comment by piegamesde** [comment]
@SuperSandro2000 This is not the time that you merge a PR with outstanding discussion items. (I had a commit prepared that I thought I'd pushed onto that branch, but the push failed and I missed it)

**Comment by SuperSandro2000** [comment]
Sorry, then I misunderstood your comment but what you are trying to do does not work anyway. Explanation will be in the linked PR.

**Pull request merged**